compiler/rustc_parse/src/parser/pat.rs

```diff
@@ -313,6 +313,9 @@ impl<'a> Parser<'a> {
             let pat = self.parse_pat_with_range_pat(false, None)?;
             self.sess.gated_spans.gate(sym::box_patterns, lo.to(self.prev_token.span));
             PatKind::Box(pat)
+        } else if self.check_keyword(kw::Const) {
```

Comment:
if you add the lookahead for `{` here, too, then the `let const = 42;` test can probably be reverted and works again as it did before

Reply:
I was thinking about that at some point, went ahead tried out and yeah, the test is now fixed.